### PR TITLE
[QNN] Conv2D type checking for kernel per-channel scales.

### PR DESCRIPTION
**Pull request by anijain2305**
As Title.

@FrozenGene @jackwish @shoubhik @yzhliu

### CODE REVIEW

File: src/relay/qnn/op/convolution.cc [hunk 1]
```diff
@@ -57,7 +57,9 @@ bool QnnConv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   CHECK(IsScalarType(types[2], DataType::Int(32)));    // input_zero_point
   CHECK(IsScalarType(types[3], DataType::Int(32)));    // kernel_zero_point
   CHECK(IsScalarType(types[4], DataType::Float(32)));  // input_scale
-  CHECK(IsScalarType(types[5], DataType::Float(32)));  // kernel_scale
+  // Kernel scale can be a vector of length output_channels or a scalar.
+  size_t axis = param->kernel_layout.find("O");
+  AssignType(types[4], DataType::Float(32), weight->shape[axis], reporter);  // kernel scale
```
Comment (shoubhik) [inline src/relay/qnn/op/convolution.cc]: should this be types[5]? there is a types[4] on line 59.

File: src/relay/qnn/op/convolution.cc [hunk 2]
```diff
@@ -57,7 +57,9 @@ bool QnnConv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   CHECK(IsScalarType(types[2], DataType::Int(32)));    // input_zero_point
   CHECK(IsScalarType(types[3], DataType::Int(32)));    // kernel_zero_point
   CHECK(IsScalarType(types[4], DataType::Float(32)));  // input_scale
-  CHECK(IsScalarType(types[5], DataType::Float(32)));  // kernel_scale
+  // Kernel scale can be a vector of length output_channels or a scalar.
+  size_t axis = param->kernel_layout.find("O");
```
Comment (shoubhik) [inline src/relay/qnn/op/convolution.cc]: Since it is a single char, better use 'O' instead of "O", from C++ perspective.

Comment (shoubhik) [reply src/relay/qnn/op/convolution.cc]: Also we should check what happens if there is no "O", just to be safe.

**Comment by anijain2305** [comment]
@shoubhik Good catch about the index. Comments addressed.

**Comment by jackwish** [comment]
Thank you for ping, and sorry for missed this. I had a busy week. :)

**Pull request merged**